### PR TITLE
Improve log messages related to OIDC session cookie encryption secret

### LINKED ISSUE #33532: [oidc] Clarify startup warning: `Secret key for encrypting tokens should be 32 characters long`
### Description

When starting Quarkus, the following warning message will be displayed if your client secret is less than 32 characters in length.

**Configuration**
```yml
quarkus:
  oidc:
    client-id: "example-client-id.apps.googleusercontent.com"
    credentials:
      secret: "example-client-secret"
```

**Logs**
```
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2023-05-22 16:18:03,236 WARN  [io.qua.oid.run.TenantConfigContext] (vert.x-eventloop-thread-1) Secret key for encrypting tokens should be 32 characters long
```

-----

It is unclear what the significance of this warning is or what action should be taken by users, as a client secret is usually something generated by an external Identity Provider that you cannot customize.

Looking at #32192, and specifically the `TenantConfigContext#createTokenEncSecretKey` method, it appears that Quarkus is using the client secret (`quarkus.oidc.credentials.secret`) to encrypt the OIDC session cookie. However, this is not mentioned in the [OIDC configuration reference](https://quarkus.io/guides/security-oidc-configuration-properties-reference#quarkus-oidc_quarkus.oidc.credentials.secret) or the [Migration Guide](https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.0#session-cookie-is-encrypted-by-default) and it's unclear whether this is intentional or if it should be referencing `quarkus.oidc.token-state-manager.encryption-secret` (as mentioned in the migration guide).

https://github.com/quarkusio/quarkus/blob/d73267b857555515f7e55b65f11e89baa825457d/extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantConfigContext.java#L85-L87

### Implementation ideas

Add further clarifications to the warning message so that users can easily tell what the issue is and how they can fix it.

### PR DESCRIPTION
**Pull request by sberyozkin**
Fixes #33532.

@geoand Can you please review and help to improve the wording ? No OIDC logic has been affected, only more log messages are added, and the warning is issued when the encryption key strength is low.

@rgmz FYI. 

Note, OIDC code could've avoided falling back to the configured secrets and just generated a random secret but that would increase the risk of the applications being broken (after restarts in devmode, multiple pods) and force users disable the encryption in any case. Seeing a log message is less intrusive. 

What I have also done is to log a warning when the key strength is very low (less than 16 chars) but only debug when the key is strong enough - that should improve the experience as the secrets generated by providers will most definitelybe at least 16 characters long. It looks like the warning message can frustrate quite a few users - the whole session cookie encryption is a hardening fix, so for users running securely over HTTPS or internal networks it can be a bit annoying having to go and disable the encryption just to get rid of the warning.

I also did not go into all the details how the fallback works, as the client secret can be configured in one of the 2 ways, and then if it is a `client_jwt_secret` authentication between Quarkus and OIDC then it is a jwt secret property, and listing all of that would make a difficult to follow doc. But I added more log messages to show how the fallback might work.

### CODE REVIEW

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java [hunk 1]
```diff
@@ -372,15 +372,17 @@ public enum Strategy {
         public boolean splitTokens;
 
         /**
-         * Requires that the tokens are encrypted before being stored in the cookies.
+         * Requires that the session cookie storing the tokens is encrypted.
```
Comment (geoand) [inline extensions/oidc/runtime/src/main/java/io/quarkus/oidc/OidcTenantConfig.java:375]:
```suggestion
         * Mandates that the session cookie that stores the tokens is encrypted.
```

**Comment by sberyozkin** [comment]
I'll add another comment as proposed at https://github.com/quarkusio/quarkus/issues/33532#issuecomment-1558079944 after all a bit later :-)

**Comment by sberyozkin** [comment]
Hi @gsmet @geoand, I'm pinging Pedro to have a quick look, but  if you think it is safe, please approve and may be it might get to 2.13. It is a pure doc clarification and log noise minimization for the out of the box OIDC user experience, no OIDC logic is affected

**Review by gsmet — DISMISSED** [review]

File: extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantConfigContext.java [hunk 2]
```diff
@@ -68,21 +68,39 @@ private static SecretKey createPkceSecretKey(OidcTenantConfig config) {
...
+                    LOG.warn("Secret key for encrypting tokens in a session cookie is missing, auto-generating it");
                     KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
                     keyGenerator.init(256);
                     return keyGenerator.generateKey();
                 }
                 byte[] secretBytes = encSecret.getBytes(StandardCharsets.UTF_8);
                 if (secretBytes.length < 32) {
-                    LOG.warn("Secret key for encrypting tokens should be 32 characters long");
+                    String errorMessage = "Secret key for encrypting tokens in a session cookie should be at least 32 characters long"
+                            + " for the strongest cookie encryption be produced."
```
[the remark this answers is not in the document]
Comment (sberyozkin) [reply extensions/oidc/runtime/src/main/java/io/quarkus/oidc/runtime/TenantConfigContext.java:94]: Hey Guillaume, yeah, I was not sure how to express in a simple way that the key matching the algorithm key size requirements will help producing the strongest encryption, and inevitably I got confused, thanks for fixing it

**Review by pedroigor — APPROVED** [review]

**Pull request merged**